doc/source/reference/c-api.iterator.rst

```diff
@@ -490,7 +490,10 @@ Construction and Destruction
 
             Indicate how the user of the iterator will read or write
             to ``op[i]``.  Exactly one of these flags must be specified
-            per operand.
+            per operand. If ``NPY_ITER_READWRITE`` or ``NPY_ITER_WRITEONLY``
+            is used for a user-provided operand, ``NpyIter_Close`` must be
+            called to possibly resolve copying the data back to the original
+            array when iteration is complete.
```

Comment:
Maybe we should just say it should always be called.

Reply:
I am not so sure. Why incur the extra function call?

Reply:
I'm not sure either. But it seems safer to always recommend it, and it's a negligible performance hit.

I'm just worried that sometimes it's hard to tell whether you need to call `NpyIter_Close` or not - it's easy to write code without it that seems OK and doesn't give any warnings, eg if you always give it the right dtype arrays in yout unit tests, but one day 2 years later you happen to pass it an array that nditer decides needs to be copied, and then you get a surprise deprecation warning.

Also, in the doc for `NpyIter_Close` you say it "Must be called".

Reply:
fixed in 3f2ba8f